code/script.js

```diff
@@ -231,95 +235,176 @@ const setSecret = () => {
     }
   }
 
+  else if (category === 'eyes') {
+    if (keep) {
+      alert(`Yes, the person has ${value} eyes! Keep all people with ${value} eyes.`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(`No, the person doesnt have ${value} eyes! Remove all people with ${value} eyes.`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
+
+  }
+
+
+
+
   // Determine what is the category
   // filter by category to keep or remove based on the keep variable.
-  /* 
+  /*
     for hair and eyes :
       charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
       or
       charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
+ 
     for accessories and other
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
       or
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
   */
-
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard();
 }
 
+
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
-  // remember the confirm() ?
+  const confirmGuess = confirm(`Do you think ${personToConfirm} is the secret person?`)
+  
   // If the player wants to guess, invoke the checkMyGuess function.
+  if (confirmGuess === true) {
+    checkMyGuess(personToConfirm)
+  } else {
+    return false
+  }
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
   // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
+  if (personToCheck === secret.name) {
+// 2. Set a Message to show in the win or lose section accordingly
+    winOrLoseText.innerHTML = `Good job, Your guess on ${personToCheck} is the correct person.`
+    
+  } else {
+    winOrLoseText.innerHTML = `Guess again, ${personToCheck} is not the person.`
+    
+  }
+  
   // 3. Show the win or lose section
+  winOrLose.style.display = 'flex'
   // 4. Hide the game board
+  board.style.display = 'none'
+  
 }
 
 // Invokes the start function when website is loaded
 start()
 
 // All the event listeners
-restartButton.addEventListener('click', start)
+
+// restarts the game at any moment on the start page
+restartButton.addEventListener('click', () => window.location.reload())
+// Change the default value for each time we select a value in the dropdown
+questions.addEventListener("change", () => selectQuestion())
+// Will let you ask question/s
+findOutButton.addEventListener('click', checkQuestion)
+
+//PlayAgainButton restarts the game
+playAgainButton.addEventListener('click', () => window.location.reload())
+
```

Comment:
Nice comments! They help to understand what they do.
Another way to approach it could be invoking the funktion inside the event listener. Probably like this: 
``restartButton.addEventListener('click', start)``
or 
``questions.addEventListener('change', selectQuestion)``